data_cache.py

```diff
@@ -0,0 +1,117 @@
+import datetime
+from pymongo import MongoClient, GEO2D
+from geopy.distance import vincenty
+
+
+class DataCache(object):
+    """
+    Maintains a connection to a MongoDB database used for caching location-based data,
+    and a set of methods for interacting with that database.
+
+    Attributes:
+        mongo_uri: A string that tells what MongoDB to use for the location cache.
+        db: DB to use.
+        collection_name: String name of current collection.
+        collection: Collection within DB where location data is stored.
+        distance_threshold: A float that the nearest cache hit must be within.
+        time_threshold: Longest time cache data is valid in database.
+        client: Mongo client initialized with mongo_uri.
+    """
+
+    def __init__(self, mongo_uri, db_name, collection_name, distance_threshold=10.0, time_threshold=60):
```

Comment:
Since you already provide default values for distance threshold and time threshold in `main.py`, I think these defaults are not useful.

Reply:
@kapil1garg My review comment got hidden by some of my short fixes